File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        int cnt=0;
        string ans="";
        for(char &x:s){
            if(x=='('){
                if(cnt>0) ans+='(';
                cnt++;
            }else{
                cnt--;
                if(cnt>0) ans+=')';
            }
        }
        return ans;
    }
};

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cs
public class Solution {
    public string RemoveOuterParentheses(string s) {
        StringBuilder ans=new StringBuilder();
        int cnt=0;
        foreach(char x in s){
            if(x=='('){
                if(cnt>0) ans.Append('(');
                cnt++;
            }else{
                cnt--;
                if(cnt>0) ans.Append(')');
            }
        }
        return ans.ToString();
    }
}